s2and/extentions/featurization/featurizer.py

```diff
@@ -1,37 +1,40 @@
-import json 
+import json
 import numpy as np
-from typing import Callable, Dict, Tuple, Union
+from os.path import join
+from typing import Dict, Tuple, Union, List, Any
 from s2and.extentions.utils import load_dataset, load_signatures
+from s2and.extentions.featurization.operations import Registry
+
 
 class Featurizer():
     '''
     Class for creating numpy arrays containg the features for each signature pair
     '''
 
     def __init__(
-        self, 
-        dataset_name: str, 
-        featurizing_function: Callable, 
-        default_embeddings: bool = True,
-        embeddings_path: Union[None, str] = None
+        self,
+        dataset_name: str,
+        features: List[Dict[str, str]],
+        embeddings_dir: Union[None, str] = None
     ) -> None:
         """
         Initializes Featurizer object, loading datasets and signatures.
         If parse_specter = true, it parses the provided specter embeddings
         for them to be used in the featurization process.
         """
-        self.default_embeddings = default_embeddings
         self.dataset = load_dataset(dataset_name)
         self.extended_signatures = load_signatures(dataset_name)
-        self.featurizing_function = featurizing_function
-        if not default_embeddings:
-            with open(embeddings_path) as f:
+        self.features = features
+        if embeddings_dir is not None:
+            with open(join(embeddings_dir, dataset_name, f'{dataset_name}_embeddings.json')) as f:
```

Comment:
Generic name, may fail to load a huge file

Reply:
Unfortunately we need to create an dictionary of signature_id -> embedding. This needs to be loaded into memory.

Reply:
Done with renaming